plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -41,10 +41,21 @@
       "com.google.cloud.tools.eclipse.sdk/enable.managed.cloud.sdk";
 
   // To be able to write tests for the managed Cloud SDK feature, which is disabled at the moment.
-  @VisibleForTesting
-  public static boolean forceManagedSdkFeature;
+  @VisibleForTesting public static boolean forceManagedSdkFeature;
+
+  @VisibleForTesting public static CloudSdkManager singleton;
+
+  public static synchronized CloudSdkManager getDefault() {
+    if (singleton == null) {
+      singleton = new CloudSdkManager();
```

Comment:
Can we make the constructor explicit and not public? As written this is not a true singleton. Other instances can be created.